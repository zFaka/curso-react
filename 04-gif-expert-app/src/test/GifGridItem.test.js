import React from 'react';
import {shallow} from 'enzyme';
import GifGridItem from '../components/atoms/GifGridItem';

describe('GifExpertApp', () => {

  const title = "Titulos";
  const url = "https://www.google.com";
  const renderOutput = shallow(<GifGridItem title={title} url={url} />);

  test('Should show component', () => {

    expect(renderOutput).toMatchSnapshot();

  })

  test('Should be equal img and alt props', () => {

    const img = renderOutput.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);

  })

  test('Should have a p with the title inside', () => {

    const p = renderOutput.find('p');

    expect(p.text().trim()).toBe(title)

  })

  test('Should have animate__fadeIn', () => {

    const div = renderOutput.find('div');
    const className = div.prop('className');
    
    expect(className.includes('animate__fadeIn')).toBe(true);

  })

})
