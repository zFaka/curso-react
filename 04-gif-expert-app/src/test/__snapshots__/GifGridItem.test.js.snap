// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifExpertApp Should show component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulos"
    src="https://www.google.com"
  />
  <p>
    Titulos
  </p>
</div>
`;
